lineapy/graph_reader/utils.py

```diff
@@ -15,3 +20,42 @@ def _is_import_node(graph: Graph, node_id: LineaID) -> bool:
                 lookup_node_name = lookup_node.__dict__.get("name", "")
                 return lookup_node_name in ["l_import", "getattr"]
     return False
+
+
+def get_db_artifacts_from_artifactdef(
```

Comment:
Does this belong in graph_reader/utils.py or perhaps as a staticmethod method for LineaArtifact given its just a batched version of `get_artifact_from_def`? 

Also get_db_artifacts might not be the right name. I thought it implied we were fetching artifactORMs or something ...

Reply:
Agree on the name change to not confuse it with ORM objects. I believe there was a handy chart of standardized python function names. We can start sticking to that. I'll see if I can link it.

as for where this should belong, i want to leave it here because:
1) there is no other usecase of using lineaartifactdef to initialize lineaartifact anywhere. 
2) db class is made as a wrapper between ORM and view objects or lineaIDs and ORM objects. lineaartifactdef seems to be something else entirely. 
3) bulk-query optimization using sqlalchemy isnt very hot. So we might need to write our own custom thing and run it using the db class as a native SQL query. This means it probably will remain outside db class anyway.

Reply:
lineaartifactdef is how we should represent the "user input" when they specify an artifact. 

Right now in our code its a mix of Tuple[str, Optional[int]] and lineaartifactdef. 
Ideally, we should be converting these tuples immediately into artifact defs. 

The later when we need the actual artifact in the code, we make calls (like to this function) to convert it. 

I can see more use cases elsewhere where a user may want to specify an artifact by name and version and we woul d need to fetch LineaArtifact to support their user journey.
For example off the top of my head, if they wanted to find an artifact through some Artifact Store search UI by specifying name/version tuples and then "inspect" (something that requires info from db) the artifact they found. 

I agree this shouldnt be in db class, but I think as a static method of LineaArtifact under `linea/api/models` should be worth considering.

Reply:
synced up, first piece is taken care of, will create tickets to consolidate artifact creation helper functions/methods.